api/agent/lb_agent.go

```diff
@@ -177,8 +185,41 @@ func (a *lbAgent) Submit(callI Call) error {
 	statsDequeue(ctx)
 	statsStartRun(ctx)
 
-	err = a.placer.PlaceCall(a.rp, ctx, call)
-	return a.handleCallEnd(ctx, call, err, true)
+	errPlace := make(chan error, 1)
+
+	call.w = &AckSyncResponseWriter{
+		origin: call.ResponseWriter(),
+		acked:  make(chan struct{}, 1),
+	}
+	isAckSync := call.Type == models.TypeAcksync
+	rw := call.w.(*AckSyncResponseWriter)
+	// change the context if it is a acksync call
+	if isAckSync {
+		ctx = common.BackgroundContext(ctx)
+		// We don't want this to run indefinetely we need to guard this context
+		var cancel func()
+		ctx, cancel = context.WithTimeout(ctx, time.Duration(60+call.Timeout))
+		defer cancel()
+	}
+
+	go a.spawnPlaceCall(ctx, call, errPlace)
+
+	for {
+		select {
+		case err := <-errPlace:
+			return err
+		case <-rw.acked:
+			// if it is an acksync we return immediately otherwise we ignore the ack
+			if isAckSync {
+				return nil
+			}
+		}
+	}
+}
+
+func (a *lbAgent) spawnPlaceCall(ctx context.Context, call *call, errCh chan error) {
+	err := a.placer.PlaceCall(a.rp, ctx, call)
+	errCh <- a.handleCallEnd(ctx, call, err, true)
```

Comment:
does the ack channel change the semantics for when we should call handleCallEnd?
is the goroutine profile here different when calls are async vs sync? (ie will we pile up goroutines we otherwise would not?)

sorry for questions, I'm less familiar here, they came to mind I'm not asking for answers necessarily, but maybe they're useful to think about...